src/core/server/saved_objects/migrationsv2/integration_tests/rewriting_id.test.ts

```diff
@@ -0,0 +1,240 @@
+            layout: {
+              type: 'json',
+            },
+          },
+        },
+        loggers: [
+          {
+            name: 'root',
+            appenders: ['file'],
+          },
+        ],
+      },
+    },
+    {
+      oss: true,
+    }
+  );
+}
+
+describe('migration v2', () => {
+  let esServer: kbnTestServer.TestElasticsearchUtils;
+  let root: Root;
+
+  beforeAll(async () => {
+    await removeLogFile();
+  });
+
+  afterAll(async () => {
+    if (root) {
+      await root.shutdown();
+    }
+    if (esServer) {
+      await esServer.stop();
+    }
+
+    await new Promise((resolve) => setTimeout(resolve, 10000));
+  });
+
+  it('rewrites id deterministically for SO with namespaceType: "multiple" and "multiple-isolated"', async () => {
+    const migratedIndex = `.kibana_${pkg.version}_001`;
+    const { startES } = kbnTestServer.createTestServers({
+      adjustTimeout: (t: number) => jest.setTimeout(t),
+      settings: {
+        es: {
+          license: 'trial',
+          // original SO:
+          // [
+          //   { id: 'foo:1', type: 'foo', foo: { name: 'Foo 1 default' } },
+          //   { id: 'spacex:foo:1', type: 'foo', foo: { name: 'Foo 1 spacex' }, namespace: 'spacex' },
+          //   {
+          //     id: 'bar:1',
+          //     type: 'bar',
+          //     bar: { nomnom: 1 },
+          //     references: [{ type: 'foo', id: '1', name: 'Foo 1 default' }],
+          //   },
+          //   {
+          //     id: 'spacex:bar:1',
+          //     type: 'bar',
+          //     bar: { nomnom: 2 },
+          //     references: [{ type: 'foo', id: '1', name: 'Foo 1 spacex' }],
+          //     namespace: 'spacex',
+          //   },
+          // ];
+          dataArchive: Path.join(__dirname, 'archives', '7.13.0_so_with_multiple_namespaces.zip'),
+        },
+      },
+    });
+
+    root = createRoot();
+
+    esServer = await startES();
+    const coreSetup = await root.setup();
+
+    coreSetup.savedObjects.registerType({
+      name: 'foo',
+      hidden: false,
+      mappings: { properties: { name: { type: 'text' } } },
+      namespaceType: 'multiple',
+      convertToMultiNamespaceTypeVersion: '8.0.0',
+    });
+
+    coreSetup.savedObjects.registerType({
+      name: 'bar',
+      hidden: false,
+      mappings: { properties: { nomnom: { type: 'integer' } } },
+      namespaceType: 'multiple-isolated',
+      convertToMultiNamespaceTypeVersion: '8.0.0',
+    });
+
+    const coreStart = await root.start();
+    const esClient = coreStart.elasticsearch.client.asInternalUser;
+
+    const migratedDocs = await fetchDocs(esClient, migratedIndex);
+
+    // each newly converted multi-namespace object in a non-default space has its ID deterministically regenerated, and a legacy-url-alias
+    // object is created which links the old ID to the new ID
+    const newFooId = uuidv5('spacex:foo:1', uuidv5.DNS);
+    const newBarId = uuidv5('spacex:bar:1', uuidv5.DNS);
```

Comment:
Is it worth internally exporting `deterministicallyRegenerateObjectId` from `src/core/server/saved_objects/migrations/core/document_migrator.ts` to be sure to use the actual implementation?